Month2/week1/Month2-Project Week 1-Remembering the Bases/VideoGame/Steam.cpp

```diff
@@ -0,0 +1,408 @@
+		{
+			std::cout << CategoryIndex << " " << CategoryContainer.GetCategory(CategoryIndex).GetCategoryTitle() << std::endl;
+		}
+		std::cout << "What do you want to do?" << std::endl;
+		std::cout << "1. Create Category" << std::endl;
+		std::cout << "2. Eliminate Category" << std::endl;
+		std::cout << "3.Back to the Main menu" << std::endl;
+		int ChoiceAction;
+		std::cin >> ChoiceAction;
+		while (!std::cin.good() || Choice <= 0 || Choice > 3) 
+		{
+			std::cin.clear();
+			std::cin.ignore(1000,'\n');
+			std::cout << "Invalid input, please enter a new Choice:" << std::endl;
+			std::cin >> ChoiceAction;
+		}
+		if (ChoiceAction == 1)
+		{
+			std::cin.clear();
+			std::cin.ignore(1000, '\n');
+			system("CLS");
+			std::cout << "Please enter the name of your new category:" << std::endl;
+			std::string Title;
+			std::getline(std::cin, Title);
+			while (!std::cin.good())
+			{
+				std::cin.clear();
+				std::cin.ignore(1000, '\n');
+				std::cout << "Invalid input or title for a category, Please enter a new name for the category:" << std::endl;
+				std::getline(std::cin, Title);
+			}
+			if (CategoryContainer.GetCategoryCount() >= 5)
+			{
+				std::cout << "Sorry, you cannot add more categories" << std::endl;
+				std::cout << "you can get back to the Category Menu menu Presssing Enter or Presss 2 for go back to the main menu" << std::endl;
+				std::cin >> Choice;
+				if (!std::cin.good() || Choice >= 3 || Choice <= 1)
+				{
+					std::cin.clear();
+					std::cout << "Invalid input, Please enter your new Choice:" << std::endl;
+					std::cin >> Choice;
+				}
+				else if (std::cin.peek() == '\n' || Choice == '\n')
+				{
+					ActiveCommand = EMenuCommand::Categories;
+					return;
+				}
+				if (Choice == 2)
+				{
+					ActiveCommand = EMenuCommand::MainMenu;
+				}
+			}
+			FCategory NewCategory(Title);
+			CategoryContainer.AddCategory(NewCategory);
+			std::cout << "Category Created" << std::endl;
+			std::cout << "What do you want to do?" << std::endl;
+			std::cout << "1.Create other category" << std::endl;
+			std::cout << "2.Back to the main Menu" << std::endl;
+			std::cin >> Choice;
+			if (!std::cin.good() || Choice >= 3 || Choice <= 0) 
+			{
+				std::cin.clear();
+				std::cin.ignore(1000, '\n');
+				std::cout << "Invalid input, please enter a new Choice";
+				std::cin >> Choice;
+			}
+		}
+		else if (ChoiceAction == 2)
+		{
+			int Choice;
+			if (CategoryContainer.GetCategoryCount() == 0)
+			{
+				std::cout << "you dont have categories to edit " << std::endl;
+				std::cout << "Presss 2 for back to de Main menu" << std::endl;
+				std::cout << "Presss enter to go back to the previous menu" << std::endl;
+				std::cin >> Choice;
+				if (std::cin.peek() == '\n' || Choice == '\n')
+				{
+					ActiveCommand = EMenuCommand::Categories;
+					return;
+				}
+			}
+			else
+			{
+				for (int CategoriesIndex = 0; CategoriesIndex < 5; CategoriesIndex += 1)
```

Comment:
You're hardcoding the max index for your categories, even though you have a variable for that in the category container. If your category count changed in the category container, your code would have a bug here.